src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -202,6 +215,11 @@ private static bool CanBindType(ModelMetadata modelMetadata)
                 return false;
             }
 
+            if(modelMetadata.IsCollectionType)
```

Comment:
This gets hit when you have a collection type that can't be handled by `GenericModelBinder`. In theory there would be more repro cases that behave like #2793